exporter/containerimage/writer.go

```diff
@@ -177,6 +199,37 @@ func (ic *ImageWriter) exportLayers(ctx context.Context, compressionType compres
 	return out, nil
 }
 
+func (ic *ImageWriter) ensureCompressionType(ctx context.Context, idxDesc ocispec.Descriptor, remotes []solver.Remote, compressionType compression.Type) (*ocispec.Descriptor, error) {
+	var layerConvertFunc converter.ConvertFunc
+	switch compressionType {
+	case compression.Any:
+		// If no compression type is specified, we don't need to convert it.
+		// lazy layers remain lazy.
+		return &idxDesc, nil
+	case compression.Uncompressed:
+		layerConvertFunc = uncompress.LayerConvertFunc
+	case compression.Gzip:
+		layerConvertFunc = gzipLayerConvertFunc
```

Comment:
Can we move this converter func to the main repo later?

Reply:
Yes, I will that.